packages/wallet/src/W3mFrame.ts

```diff
@@ -0,0 +1,107 @@
+import { W3mFrameConstants } from './W3mFrameConstants.js'
+import { W3mFrameSchema } from './W3mFrameSchema.js'
+import { W3mFrameHelpers } from './W3mFrameHelpers.js'
+import type { W3mFrameTypes } from './W3mFrameTypes.js'
+
+// -- Sdk --------------------------------------------------------------------
+export class W3mFrame {
+  private iframe: HTMLIFrameElement | null = null
+
+  private projectId: string
+
+  private rpcUrl = W3mFrameHelpers.getBlockchainApiUrl()
+
+  public frameLoadPromise: Promise<void>
+
+  public frameLoadPromiseResolver:
+    | {
+        resolve: (value: undefined) => void
+        reject: (reason?: unknown) => void
+      }
+    | undefined
+
+  public constructor(projectId: string, isAppClient = false) {
+    this.projectId = projectId
+    this.frameLoadPromise = new Promise((resolve, reject) => {
+      this.frameLoadPromiseResolver = { resolve, reject }
+    })
+
+    // Create iframe only when sdk is initialised from dapp / web3modal
+    if (isAppClient) {
+      this.frameLoadPromise = new Promise((resolve, reject) => {
+        this.frameLoadPromiseResolver = { resolve, reject }
+      })
+      const iframe = document.createElement('iframe')
+      iframe.id = 'w3m-iframe'
+      iframe.src = `${W3mFrameConstants.SECURE_SITE_SDK}?projectId=${projectId}`
+      iframe.style.position = 'fixed'
+      iframe.style.zIndex = '999999'
+      iframe.style.display = 'none'
+      iframe.style.opacity = '0'
+      iframe.style.borderRadius = `clamp(0px, var(--wui-border-radius-l), 44px)`
+      document.body.appendChild(iframe)
+      this.iframe = iframe
+      this.iframe.onload = () => {
+        this.frameLoadPromiseResolver?.resolve(undefined)
+      }
+      this.iframe.onerror = () => {
+        this.frameLoadPromiseResolver?.reject()
+      }
+    }
+  }
+
+  // -- Networks --------------------------------------------------------------
+  get networks(): Record<number, W3mFrameTypes.Network> {
+    const data = [
+      1, 5, 11155111, 10, 420, 42161, 421613, 137, 80001, 42220, 1313161554, 1313161555, 56, 97,
```

Comment:
Would it be proper to use `PresetsUtil.WalletConnectRpcChainIds` instead of redefining the chains array here?

Reply:
Will do this for beta, as this particular preset bit needs to be taken out into common package now (to not include scaffold bits into wallet package)